### PR TITLE
Upgrade ts version and webpack version 

### PR DESCRIPTION
**Pull request by mt-bt**
In preparation of https://github.com/palantir/conjure-typescript/pull/181 

As discussed with @ferozco let's release these changes first

## Before this PR
outdated version of TS and webpack version

## After this PR
updated version of TS/webpack

==COMMIT_MSG==
Upgrade typescript/webpack versions
==COMMIT_MSG==

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -35,21 +23,26 @@ module.exports = {
...
+        '^.+\\.(js|jsx|ts|tsx)$': 'ts-jest',
     },
     transformIgnorePatterns: [
-        "/node_modules/(?!(lodash-es)/)",
+        '/node_modules/(?!(lodash-es)/)',
     ],
     verbose: true,
-};
+    preset: 'ts-jest/presets/js-with-ts',
+}
```
Comment (mt-bt) [inline jest.config.js:48]: ran the jest migration script to update this file - it seems to work fine!

File: src/commands/generate/__tests__/cliTest.ts [hunk 2]
```diff
@@ -186,8 +186,8 @@ describe("generate command", () => {
         expect(fs.existsSync(path.join(outDir, "node_modules/typescript/bin/tsc")));
     });
 
+    jest.setTimeout(10000);
```
Comment (ferozco) [inline src/commands/generate/__tests__/cliTest.ts]: why do we move this out of the test case? seems like other test cases still set the timeout explicitly

Comment (mt-bt) [reply src/commands/generate/__tests__/cliTest.ts]: I had a similar issue as https://github.com/facebook/jest/issues/11543 it's simply ignored, I should probably spend more time looking into it

**Comment by ferozco** [comment]
👍

File: src/commands/generate/__tests__/cliTest.ts (hunk 2 above)
Comment (mt-bt) [reply src/commands/generate/__tests__/cliTest.ts]: lifted this up to the top level describe and removed from each individual test, are you ok with that?

**Pull request merged**